src/app/tests/suites/commands/interaction_model/InteractionModel.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "InteractionModel.h"
+
+CHIP_ERROR InteractionModel::ReadAttribute(const char * identity, chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                           chip::AttributeId attributeId, bool fabricFiltered)
+{
+    chip::DeviceProxy * device = GetDevice(identity);
+    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    chip::app::AttributePathParams attributePathParams[1];
+    attributePathParams[0].mEndpointId  = endpointId;
+    attributePathParams[0].mClusterId   = clusterId;
+    attributePathParams[0].mAttributeId = attributeId;
```

Comment:
This is making assumptions about how `AttributePathParams` encodes wildcards, no?  We should not introduce that coupling.  Ideally the args here would be `Optional`, but if they use inline encoding of wildcards that should be documented and the magic values checked for before assigning to `attributePathParams`.

Reply:
> This is making assumptions about how `AttributePathParams` encodes wildcards, no? We should not introduce that coupling. Ideally the args here would be `Optional`, but if they use inline encoding of wildcards that should be documented and the magic values checked for before assigning to `attributePathParams`.

I understand the your position and I can update the code to add comments, chip::Optional, etc...  But I suspect that there is a lack of clarity in the spec too and I want to make sure that "comments" are going to the right place ;)

To be honest, I was not thinking that I was making assumptions about some of the stack internals, but that I'm following the spec. The spec is unclear about what is a "wildcard" and I would like it to be clearer.

First, the spec defining a "Wildcard Path" does not even spoke of the attributes/events parts of wildcards: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Interaction-Model.adoc#214-wildcard-path (unless "cluster" implicitly includes those, which is unclear to me).

Secondly I originally made use of what is called "invalid" values in the code because of the Manufacturer Extensions spec at https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Data-Model-Namespace.adoc
In this part of the spec there is the mention of "a special wildcard value". And the "wildcard" value for MEI prefix and MEI suffix is "0xFFFF" and I (maybe stupidly) translate that to "0xFFFF" == "special wildcard value" which I ends up using for wildcard features in `chip-tool`. And which I expanding here.

Third, `AttributePathParams` `EventPathParams` use those values in method called `HasWildcard*Id` (https://github.com/project-chip/connectedhomeip/blob/4967c0e0a51e3e0f0216c551d35b4cbde5aea7e4/src/app/AttributePathParams.h#L62)

So. If I try to mix up all that. My understanding of what the spec says is that:
  * `0xFFFF` is a wildcard value for the "attributes"/"events"  bits
  *  no special value is defined for the endpoint cluster bits 
     * It seems that there was some tries to include those values into the spec at https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/4976#discussion_r813598200 and https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/4976#discussion_r813598446

Reply:
For the spec side, see https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/4229

The basic structure of the spec is that in the Interaction Model specification there is a concept of "wildcard" but no description of how a wildcard is represented.

On the wire, the representation of wildcards is defined by the relevant encoding specification (in this case chapter 10 of the Matter spec), which does not use the "wildcard" MEI values.

MEIs do define a "wildcard" value but it's not used in that fashion anywhere in the spec per the spec issue above.

`AttributePathParams` and `EventPathParams` do use the "wildcard" MEI values internally because those are not valid values for actual attribute id or event id or cluster id (note that the endpoint id is not a MEI, but also uses an invalid value to represent "wildcard").  That's an implementation details of those params structs; the way we should probably set wildcards on them is by either not setting the relevant field at all or by calling the `SetWildcard*` bits that #17417 is introducing.

I mean, setting the invalid values being used will work, but if we ever change to using any of the other invalid values for some reason, or some other representation, things would break.  So best to not depend on the actual internal representation if we can avoid it.

Reply:
> I have updated the code to not set the field if the target id is invalid.

I'm planning to update this code soon since currently the test suite does not makes use of wildcards but it will need to so. Also there may part some duplication between `examples/chip-tool/commands/clusters` and this interaction model module that I would like to factorize. That would be a good place to document as it will affects both `chip-tool` command and the test suites.